cmd/dockerd/hack/malformed_host_override_test.go

```diff
@@ -0,0 +1,184 @@
+// +build !windows
+
+package hack
+
+import (
+	"bytes"
+	"io"
+	"net"
+	"strings"
+	"testing"
+)
+
+func TestHeaderOverrideHack(t *testing.T) {
+	client, srv := net.Pipe()
+	req := []byte("GET /foo\nHost: /var/run/docker.sock\nUser-Agent: Docker\r\n\r\n")
+	read := make([]byte, 4096)
+
+	l := MalformedHostHeaderOverrideConn{client, true}
+	go func() {
+		if _, err := srv.Write(req); err != nil {
```

Comment:
No need to check for errors in these write calls. Can just `srv.Write(req)`  and then `srv.Cloe()`